src/chronify/time_series_mapper_representative.py

```diff
@@ -0,0 +1,131 @@
+from sqlalchemy import Engine, MetaData, Table, select
+
+import pandas as pd
+
+from chronify.sqlalchemy.functions import read_database, write_database
+from chronify.models import TableSchema
+from chronify.exceptions import MissingParameter, ConflictingInputsError
+from chronify.time_series_mapper_base import TimeSeriesMapperBase
+from chronify.utils.sqlalchemy_view import create_view
+from chronify.time_models import RepresentativePeriodTimeGenerator
+
+
+class MapperRepresentativeTimeToDatetime(TimeSeriesMapperBase):
+    def __init__(
+        self, engine: Engine, metadata: MetaData, from_schema: TableSchema, to_schema: TableSchema
+    ) -> None:
+        self._engine = engine
+        self._metadata = metadata
+        self.from_schema = from_schema
+        self.to_schema = to_schema
+        self._generator = RepresentativePeriodTimeGenerator(from_schema.time_config)
+
+    def check_schema_consistency(self) -> None:
+        """Check that from_schema can produce to_schema."""
+        self._check_table_column_producibility()
+        self._check_schema_measurement_type_consistency()
+
+    def _check_table_column_producibility(self) -> None:
+        """Check columns in destination table can be produced by source table."""
+        available_cols = self.from_schema.list_columns() + [self.to_schema.time_config.time_column]
+        final_cols = self.to_schema.list_columns()
+        if diff := set(final_cols) - set(available_cols):
+            msg = f"Source table {self.from_schema.time_config.name} cannot produce the columns: {diff}"
+            raise ConflictingInputsError(msg)
+
+    def _check_schema_measurement_type_consistency(self) -> None:
+        """Check that measurement_type is the same between schema."""
+        from_mt = self.from_schema.time_config.measurement_type
+        to_mt = self.from_schema.time_config.measurement_type
+        if from_mt != to_mt:
+            msg = f"Inconsistent measurement_types {from_mt=} vs. {to_mt=}"
+            raise ConflictingInputsError(msg)
+
+    def _check_source_table_has_time_zone(self) -> None:
+        """Check source table has time_zone column."""
+        if "time_zone" not in self.from_schema.time_array_id_columns:
```

Comment:
This check seems debatable. The input data could have an `id` column that uniquely identifies time arrays. Why impose an extra burden on this setting? Can it instead check for the presence of the column in the table?